Create HTML sitemap package and `allPublishedContentDates` query

B77Mills opened the pull request:
- Add resolver for allPublishedContentDates
 - Add definition for allPublishedContentDates
 - Setup new marko-web-html-sitemap package
   - Add routing & templates for new html sitemap
   - Add minor styling
   - Add fragment for publishedContent

The site repos will have to import the css & routes once the package is added to their repo.

<img width="2575" alt="Screen Shot 2021-05-28 at 9 43 17 AM" src="https://user-images.githubusercontent.com/3845869/120001212-2cf3aa80-bf99-11eb-8c7b-5b2e320492e1.png">

B77Mills committed:
Add resolver for allPublishedContentDates

B77Mills committed:
Add definition for allPublishedContentDates

B77Mills committed:
Setup new marko-web-html-sitemap package

B77Mills committed:
Add missing peer dep

B77Mills committed:
Setup eslinterrc.js file

B77Mills committed:
Add routing & templates

 - Add routing matcher for /html-sitemap/${year}/${month}/${day} and all of it child matches

 - on match of route
 — set nodes being returned to years/months/days and pass required page data like displayType(year, month, day)

On match to specific day pass corisponding before and after day to query content for that day.

B77Mills committed:
Add minor styling

B77Mills committed:
Add fragment for publishedContent

solocommand commented on packages/marko-web-html-sitemap/templates/date-list.marko:
```diff
@@ -0,0 +1,60 @@
+$ switch (displayType) {
+  case "day":
+      pageNode.description = `Select from the list of days in ${month} ${year}`;
+      break
```
You should have 4 spaces for your case block contents, not 6.

solocommand replied:
Also missing semicolons after your breaks

solocommand commented on packages/marko-web-html-sitemap/templates/date-list.marko:
```diff
@@ -0,0 +1,60 @@
+      pageNode.description = `Select from the list of months in ${year}`;
+      break
+  default:
+    pageNode.description = `Select from the list of years`;
```
This should have translation support like the inquiry package now does. ~The date format (and language) should be pulled from the site config also (it's customized on at least one client...maybe bizbash?)~
Nevermind this part -- the es-mx date format would be the same.

B77Mills replied:
@solocommand I am going to do a second PR for adding in lang support

solocommand commented on packages/marko-web-html-sitemap/templates/date-list.marko:
```diff
@@ -0,0 +1,60 @@
+};
+$ switch (displayType) {
+  case "day":
+      pageNode.description = `Select from the list of days in ${month} ${year}`;
```
All {siteName} stories published in {month} {year}.

B77Mills replied:
@solocommand update this to the following: 
 - All Equipment World stories published by years
 - All Equipment World stories published by months in 2003
 - All Equipment World stories published by days in November 2003
 - All Equipment World stories published on November 26, 2003

solocommand commented on packages/marko-web-html-sitemap/templates/date-list.marko:
```diff
@@ -0,0 +1,60 @@
+    pageNode.description = `Select from the list of years`;
+};
+
+$ const nameOfMonth = [
```
Use a library to get the localized version of the month name. On mundo this should be `junio` rather than `July`, etc. Or use the translation utility to get the name of the month (`MONTH_1_LABEL` or something)

B77Mills replied:
@solocommand I just used moment to format this.  will figure out lang support on second PR

solocommand commented on packages/marko-web-html-sitemap/marko.json:
```diff
@@ -0,0 +1 @@
+{}
```
Do you need this file? Pretty sure it's only required if you're defining components within the package.

solocommand commented on packages/marko-web-html-sitemap/routes/index.js:
```diff
@@ -0,0 +1,150 @@
+const ROOT_ALIAS = '/html-sitemap';
+
+module.exports = (app) => {
+  app.get(`${ROOT_ALIAS}/:year(\\d{4})/:month(\\d{2})/:day(\\d{2})`, asyncRoute(async (req, res) => {
```
Allow this to be passed in from the site:

```suggestion
module.exports = ({
  mountPoint: 'sitemap',
}) => (app) => {
  app.get(`${mountPoint}/:year(\\d{4})/:month(\\d{2})/:day(\\d{2})`, asyncRoute(async (req, res) => {
```

And in the consuming application:
```js
const router = require('@parameter1/base-cms-marko-web-html-sitemap/routes');

module.exports = (app) => {
  app.use(router({ mountPoint: 'some-wacky-sitemap-prefix' }));
};
```

B77Mills replied:
@solocommand  I couldn't get that version to resolve, but i did the following:

in the routes/index.js 
`module.exports = (app, { mountPoint } = { mountPoint: '/html-sitemap' }) => {`

if you do the following on use of that route
`htmlSitemap(app);`
The mountPoint will be  /html-sitemap
if you do
`htmlSitemap(app, { mountPoint: '/wacky-site' });`
The mountPoint will be /wacky-site instead.

solocommand commented on packages/marko-web-html-sitemap/templates/day.marko:
```diff
@@ -0,0 +1,43 @@
+
+$ const type = "sitemap";
+$ const date = `${month} ${day}, ${year}`;
+$ const title = 'Sitemap';
```
Use double quotes in template files

solocommand commented on packages/marko-web-html-sitemap/templates/day.marko:
```diff
@@ -0,0 +1,43 @@
+import queryFragment from "../graphql/fragments/sitemap-published-content";
+
+$ const { config, GAM } = out.global;
+$ const { ending, after, year, month, day } = input;
```
Do not destructure more than 3 variables at once, break these onto new lines.

solocommand commented on packages/marko-web-html-sitemap/templates/day.marko:
```diff
@@ -0,0 +1,43 @@
+            publishedAfter: new Date(after.toISOString()),
+            since: new Date(ending.toISOString()),
+            limit: 1000,
+            queryFragment
```
Trailing comma missing!

solocommand commented on packages/marko-web-html-sitemap/routes/index.js:
```diff
@@ -0,0 +1,150 @@
+    };
+    const variables = { input };
+    const { data } = await req.apollo.query({ query: ALL_PUBLISHED_CONTENT_DATES, variables });
+    const nodes = data.allPublishedContentDates.map((node) => {
```
```suggestion
    const nodes = getAsArray(data, "allPublishedContentDates").map((node) => {
```

You never know what sneaky results might come back!

solocommand commented on packages/marko-web-html-sitemap/routes/index.js:
```diff
@@ -0,0 +1,150 @@
+      const publishedContentNode = node;
+      publishedContentNode.alias = `${ROOT_ALIAS}/${node.year}`;
+      return publishedContentNode;
+    });
```
```suggestion
    const nodes = getAsArray(data, "allPublishedContentDates").map((node) => ({
      ...node,
      alias: `${mountPoint}/${node.year}`,
    }));
```

solocommand commented on packages/marko-web-html-sitemap/routes/index.js:
```diff
@@ -0,0 +1,150 @@
+      publishedContentNode.alias = `${ROOT_ALIAS}/${node.year}`;
+      return publishedContentNode;
+    });
+    return res.marko(
```
This should be under 100 chars, make it 1 line please

solocommand commented on packages/marko-web-html-sitemap/scss/index.scss:
```diff
@@ -0,0 +1,38 @@
+$primary: #000;
```
Does this need `!default` in order to be overwritten?

solocommand commented on packages/marko-web-html-sitemap/templates/date-list.marko:
```diff
@@ -0,0 +1,60 @@
+$ const { config, GAM } = out.global;
```
`GAM` shouldn't be used in this package

solocommand commented on packages/marko-web-html-sitemap/templates/date-list.marko:
```diff
@@ -0,0 +1,60 @@
+$ const { config, GAM } = out.global;
+$ const { nodes, displayType, month, year, alias } = input;
```
Too many vars, break to new lines

solocommand reviewed (changes requested):
Nice job! A few cleanup and style changes, but structurally this looks great.

B77Mills committed:
Remove empty file

B77Mills committed:
Update node map and add support function for 2 char numbers

B77Mills committed:
cleanup template formatting

B77Mills committed:
Include missing peer dep

B77Mills committed:
Add !default to allow overrides

zarathustra323 commented on packages/marko-web-html-sitemap/package.json:
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "@parameter1/base-cms-marko-web-html-sitemap",
+  "version": "2.20.0",
```
Change to `2.21.0` to match the current version

B77Mills committed:
Add ability for mountPoint to be changed

when including these routes in the site or package you can set a mount point as well

htmlSitemap(app);

VS

htmlSitemap(app, { mountPoint: '/wacky-sitemap' });

B77Mills committed:
Fix description and displaying of month name

B77Mills committed:
fix published

B77Mills committed:
Update description

B77Mills committed:
Add s to months

B77Mills committed:
update package version

B77Mills committed:
add default value for mountpoint for /html-sitemap

B77Mills committed:
pass mountPoint through to template

B77Mills committed:
add breadcrumb component

B77Mills committed:
Update paylayout and display breadcrumbs vs description

B77Mills committed:
update styling of dates being displayed and page layout

B77Mills committed:
only link title if not on the mountPoint
